### PR TITLE
MULE-18691: Add documentation informing that Parameter Groups must implement equals and hashcode.

### PR DESCRIPTION
**Pull request by SebaElizalde**

### CODE REVIEW

**Comment by SebaElizalde** [comment]
WIP, please do not merge until it has approval from @estebanwasinger

File: modules/ROOT/pages/define-parameters.adoc
Comment (ndinu) [inline modules/ROOT/pages/define-parameters.adoc]: I would say that the classes are the ones that must override the method instead of the parameters and the parameter groups.

Comment (ndinu) [inline modules/ROOT/pages/define-parameters.adoc]: I would say implement instead of override.

**Review by ndinu — DISMISSED** [review]

**Comment by SebaElizalde** [comment]
@mulesoft/team-docs this PR is now ready for review and merge. Thanks!

File: modules/ROOT/pages/define-parameters.adoc
Comment (sarathecoubian) [inline modules/ROOT/pages/define-parameters.adoc]:
```suggestion
Java classes used for POJO parameters or Parameter Groups must implement the `equals` and
```

File: modules/ROOT/pages/define-parameters.adoc
Comment (sarathecoubian) [inline modules/ROOT/pages/define-parameters.adoc]:
```suggestion
`hashCode` methods for <<static-dynamic-configs.adoc#Dynamic-Configuration-Lifecycle,Dynamic Configurations>> to function correctly.
```

File: modules/ROOT/pages/define-parameters.adoc
Comment (sarathecoubian) [inline modules/ROOT/pages/define-parameters.adoc]:
```suggestion
Even if existing configurations do not have POJO parameters, the configurations need `equals` and `hashCode` methods because future
```

File: modules/ROOT/pages/define-parameters.adoc
Comment (sarathecoubian) [inline modules/ROOT/pages/define-parameters.adoc]:
```suggestion
runtime functionality may depend on it. The following example from a class in the Database Connector that illustrates the implementation of `equals` and `hashcode` methods:
```

File: modules/ROOT/pages/define-parameters.adoc
Comment (sarathecoubian) [inline modules/ROOT/pages/define-parameters.adoc]:
```suggestion
* Implementing `equals` and `hashCode` methods
```

**Review by sarathecoubian — CHANGES_REQUESTED** [review]

**Review by sarathecoubian — APPROVED** [review]

**Pull request merged**
